src/pp-annotator/form/PrzypisEditor.tsx

```diff
@@ -85,68 +87,57 @@ export default class PrzypisEditor extends Widget {
   }
 
   /**
-   * Override parent attach function to render React form
+   * Loads the annotation and displays the edit window
+   * Returns an unresolved Promise that will be resolved/rejected when the save/cancel button is clicked.
    */
-  public attach() {
-    // Call parent function (renders PrzypisEditor.template)
-    super.attach();
-    this.updateForm({});
-  }
-
-  /**
-   * Returns an unresolved Promise that will be resolved when the save/cancel button is clicked.
-   * If load function is waited upon, it will finish only when the save/cancel button is clicked.
-   */
-  public load(annotation: annotator.IAnnotation, position: util.IPosition) {
+  public load(annotation: annotator.IAnnotation,
+              position: util.IPosition,
+              saveAction: (annotation: IAnnotation) => any) {
     this.annotation = annotation;
-    this.updateForm(annotation.fields || {});
+    this.saveAction = saveAction.bind(this);
```

Comment:
This doesn't seem like a good place to bind context. This class doesn't need to know what exactly `saveAction` does, so it shouldn't need to bind `this`.

Reply:
Good point, my mistakes.